test/GogeToken.t.sol

```diff
@@ -3,23 +3,263 @@ pragma solidity ^0.8.6;
+    //          - fix royalty distribution
+
+    // setBuyBackEnabled test
+    function test_gogeToken_setBuyBackEnabled() public {
+        // Pre-state check.
+        assertTrue(gogeToken.buyBackEnabled());
+        assertEq(gogeToken.buyBackFee(), 2);
+        assertEq(gogeToken.previousbuyBackFee(), 0);
+
+        // Disable buyBack
+        assert(dev.try_setBuyBackEnabled(address(gogeToken), false));
+
+        //Post-state check.
+        assertTrue(!gogeToken.buyBackEnabled());
+        assertEq(gogeToken.buyBackFee(), 0);
+        assertEq(gogeToken.previousbuyBackFee(), 2);
+    }
+
+    // setMarketingEnabled test
+    function test_gogeToken_setMarketingEnabled() public {
+        // Pre-state check.
+        assertTrue(gogeToken.marketingEnabled());
+        assertEq(gogeToken.marketingFee(), 2);
+        assertEq(gogeToken.previousMarketingFee(), 0);
+
+        // Disable buyBack
+        assert(dev.try_setMarketingEnabled(address(gogeToken), false));
+
+        //Post-state check.
+        assertTrue(!gogeToken.marketingEnabled());
+        assertEq(gogeToken.marketingFee(), 0);
+        assertEq(gogeToken.previousMarketingFee(), 2);
+    }
+
+    // setCakeDividendEnabled test
+    function test_gogeToken_setCakeDividendEnabled() public {
+        // Pre-state check.
+        assertTrue(gogeToken.cakeDividendEnabled());
+        assertEq(gogeToken.cakeDividendRewardsFee(), 10);
+        assertEq(gogeToken.previousCakeDividendRewardsFee(), 0);
+
+        // Disable buyBack
+        assert(dev.try_setCakeDividendEnabled(address(gogeToken), false));
+
+        //Post-state check.
+        assertTrue(!gogeToken.cakeDividendEnabled());
+        assertEq(gogeToken.cakeDividendRewardsFee(), 0);
+        assertEq(gogeToken.previousCakeDividendRewardsFee(), 10);
     }
 
-    function testFail_basic_sanity() public {
-        assertTrue(false);
+    // setTeamEnabled test
+    function test_gogeToken_setTeamEnabled() public {
+        // Pre-state check.
+        assertTrue(gogeToken.teamEnabled());
+        assertEq(gogeToken.teamFee(), 2);
+        assertEq(gogeToken.previousTeamFee(), 0);
+
+        // Disable buyBack
+        assert(dev.try_setTeamEnabled(address(gogeToken), false));
+
+        //Post-state check.
+        assertTrue(!gogeToken.teamEnabled());
+        assertEq(gogeToken.teamFee(), 0);
+        assertEq(gogeToken.previousTeamFee(), 2);
     }
 
-    function test_basic_sanity() public {
-        assertTrue(true);
+    function test_gogeToken_safeWithdraw() public {
+        mint("BUSD", address(gogeToken), 1_000 ether);
+
+        assertEq(IERC20(BUSD).balanceOf(address(gogeToken)), 1_000 ether);
+        assertEq(IERC20(BUSD).balanceOf(address(dev)), 0);
+
+        assert(dev.try_safeWithdraw(address(gogeToken), BUSD));
+
+        assertEq(IERC20(BUSD).balanceOf(address(gogeToken)), 0);
+        assertEq(IERC20(BUSD).balanceOf(address(dev)), 1_000 ether);
     }
+
```

Comment:
There is definitely some things that I missed. Things like migrate() and captureLiquidity() are tested in specific test files. Migration.t.sol was migration specific so that verifies functions related to migrate() and capturLiquidity(). getLastReceived, setAutomatedMarketMakerPair, excludeCirculatingSupply, updateCakeDividendToken, and prepareForPartnerOrExchange all covered in Royalties.t.sol and GogeDao.t.sol